### PR TITLE
[14.0] [MIG] module_auto_update

### PR DESCRIPTION
**Pull request by manengel**
Small migration work for module_auto_update for Odoo 14.0

### CODE REVIEW

**Comment by atchuthan** [comment]
Runbot failed due to 
2020-11-30 17:00:01,440 201 INFO openerp_test odoo.modules.registry: module module_auto_update: creating or updating database tables
2020-11-30 17:00:01,515 201 INFO openerp_test odoo.modules.loading: loading module_auto_update/views/ir_module_module.xml
2020-11-30 17:00:01,517 201 ERROR openerp_test odoo.tools.convert: The XML file '/home/odoo/build/OCA/server-tools/setup/module_auto_update/odoo/addons/module_auto_update/views/ir_module_module.xml' does not fit the required schema !
Traceback (most recent call last):
File "/home/odoo/odoo-14.0/odoo/tools/convert.py", line 782, in convert_xml_import
relaxng.assert_(doc)
File "src/lxml/lxml.etree.pyx", line 3501, in lxml.etree._Validator.assert_ (src/lxml/lxml.etree.c:184715)
AssertionError: Element odoo has extra content: record, line 7
2020-11-30 17:00:01,517 201 WARNING openerp_test odoo.tools.convert: /home/odoo/build/OCA/server-tools/setup/module_auto_update/odoo/addons/module_auto_update/views/ir_module_module.xml:7:0:ERROR:RELAXNGV:RELAXNG_ERR_EXTRACONTENT: Element odoo has extra content: record
2020-11-30 17:00:01,517 201 INFO openerp_test odoo.tools.convert: Install 'jingtrang' for more precise and useful validation messages.
2020-11-30 17:00:01,519 201 WARNING openerp_test odoo.modules.loading: Transient module states were reset
2020-11-30 17:00:01,520 201 ERROR openerp_test odoo.modules.registry: Failed to load registry
2020-11-30 17:00:01,520 201 CRITICAL openerp_test odoo.service.server: Failed to initialize database `openerp_test`.
Traceback (most recent call last):
File "/home/odoo/odoo-14.0/odoo/service/server.py", line 1198, in preload_registries
registry = Registry.new(dbname, update_module=update_module)
File "/home/odoo/odoo-14.0/odoo/modules/registry.py", line 89, in new
odoo.modules.load_modules(registry._db, force_demo, status, update_module)
File "/home/odoo/odoo-14.0/odoo/modules/loading.py", line 455, in load_modules
loaded_modules, update_module, models_to_check)
File "/home/odoo/odoo-14.0/odoo/modules/loading.py", line 348, in load_marked_modules
perform_checks=perform_checks, models_to_check=models_to_check
File "/home/odoo/odoo-14.0/odoo/modules/loading.py", line 221, in load_module_graph
load_data(cr, idref, mode, kind='data', package=package)
File "/home/odoo/odoo-14.0/odoo/modules/loading.py", line 69, in load_data
tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind)
File "/home/odoo/odoo-14.0/odoo/tools/convert.py", line 733, in convert_file
convert_xml_import(cr, module, fp, idref, mode, noupdate)
File "/home/odoo/odoo-14.0/odoo/tools/convert.py", line 782, in convert_xml_import
relaxng.assert_(doc)
File "src/lxml/lxml.etree.pyx", line 3501, in lxml.etree._Validator.assert_ (src/lxml/lxml.etree.c:184715)
AssertionError: Element odoo has extra content: record, line 7

File: module_auto_update/addon_hash.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+def _fnmatch(filename, patterns):
+    for pattern in patterns:
+        if fnmatch(filename, pattern):
+            return True
+    return False
+
+
+def _walk(top, exclude_patterns, keep_langs):
+    keep_langs = {l.split("_")[0] for l in keep_langs}
```
Comment (atchuthan) [inline module_auto_update/addon_hash.py]: flake8 except __init__.py................................................Failed
- hook id: flake8
- duration: 0.89s
- exit code: 1
module_auto_update/addon_hash.py:17:39: E741 ambiguous variable name 'l'

**Review by atchuthan — CHANGES_REQUESTED** [review]

**Comment by manengel** [comment]
@atchuthan https://github.com/OCA/server-tools/pull/1943#issuecomment-741621515 was resolved by deleting the line 23 `
 icon="fa-exchange"` of the file module_auto_update/views/ir_module_module.xml'.

File: module_auto_update/addon_hash.py (hunk 1 above)
Comment (manengel) [reply module_auto_update/addon_hash.py]: @atchuthan Issue has been adressed and fixed.

**Review by atchuthan — APPROVED** [review]

**Review by dsolanki-initos — APPROVED** [review]

**Comment by sbidoul** [comment]
Migration looks good to me. Merging as requested.

Note I highly recommend using [click-odoo-update](https://github.com/acsone/click-odoo-contrib#click-odoo-update-stable) instead which is much more robust and well maintained. 

/ocabot merge patch

**Pull request merged**